Jenkinsfile: use a list for extraArgs

arichardson opened the pull request:
Since https://github.com/CTSRD-CHERI/jenkins-scripts/commit/98ba5a94cb37cab7a43665dea075cbd36d4df9c7
extraArgs accepts a list as well.
Use a list to avoid manual quoting.

arichardson committed:
Jenkinsfile: catch up with latest changes

Since https://github.com/CTSRD-CHERI/jenkins-scripts/commit/98ba5a94cb37cab7a43665dea075cbd36d4df9c7,
the instanceof check is failing.
My guess is that `"--without-sdk --install-prefix=/usr $extraQemuArgs"`
is a GString instance so the instanceof check fails.

To simplify future maintenance, just change it to a list.

arichardson commented:
Will merge once CI passes.

jrtc27 commented:
Can you please update the commit message to reflect this is a cleanup that uses the new interface rather than a bug fix, given that I've fixed parseParams to work with GString?

arichardson commented:
> Can you please update the commit message to reflect this is a cleanup that uses the new interface rather than a bug fix, given that I've fixed parseParams to work with GString?

updated description, will squash-and-merge.
